raft/log.go

```diff
@@ -832,7 +835,7 @@ loop:
 	}
 
 	// Determine commit index by quorum (n/2+1).
-	sort.Sort(uint64Slice(indexes))
```

Comment:
This finds the highest index saved to a quorum. Let's say you have 5 nodes with indices of:

```
1,2,3,4,5
```

The highest index a quorum would have is `3` if you used nodes `3`, `4`, `5`. We reverse sort the numbers so that if we get fewer responses that 5 then we still pick the 3rd one. So let's say we only received four responses:

```
2,3,4,5
```

If we grabbed the 3rd one in normal sort order it'd be `4`. But the missing index is effectively zero. If we reverse sort:

```
5,4,3,2
```

then the 3rd index is `3` (which is correct).